src/inputType/RotatePanInput.ts

```diff
@@ -0,0 +1,136 @@
+import Axes from "../Axes";
+import { toAxis } from "./InputType";
+import { PanInput, PanInputOption } from "./PanInput";
+
+export class RotatePanInput extends PanInput {
+	private rotateOrigin: number[];
+	private prevAngle: number;
+	private prevQuadrant: number;
+	private lastDiff: number;
+	private coefficientForDistanceToAngle: number;
+
+	constructor(el: string | HTMLElement, options?: PanInputOption) {
+		super(el, options);
+
+		const rect = this.element.getBoundingClientRect();
+		// TODO: how to do if element is ellipse not circle.
+		this.coefficientForDistanceToAngle = 360 / (rect.width * Math.PI); // from 2*pi*r * x / 360
+		// TODO: provide a way to set origin like https://developer.mozilla.org/en-US/docs/Web/CSS/transform-origin
+		this.rotateOrigin = [rect.left + (rect.width - 1) / 2, rect.top + (rect.height - 1) / 2];
```

Comment:
@jongmoon 
Did rotatorigin respond to resize?

Shouldn't it be added at the time of the hold event as well as the constructor?

Reply:
@daybrush I didn't consider 'responsive` but your suggestion would be better.